### PR TITLE
refactor(layer load API): all errors are permanent

### PR DESCRIPTION
**Pull request by problame**
I am not aware of a case of "transient" VirtualFile errors as mentioned in https://github.com/neondatabase/neon/pull/5880

Private DM with Joonas discussing this: https://neondb.slack.com/archives/D049K7HJ9JM/p1721836424615799

### CODE REVIEW

File: pageserver/src/tenant/storage_layer/layer.rs [hunk 1]
```diff
@@ -1625,7 +1625,7 @@ pub(crate) struct DownloadedLayer {
     owner: Weak<LayerInner>,
     // Use tokio OnceCell as we do not need to deinitialize this, it'll just get dropped with the
     // DownloadedLayer
-    kind: tokio::sync::OnceCell<anyhow::Result<LayerKind>>,
+    kind: tokio::sync::OnceCell<LayerKind>,
```
Comment (koivunej) [inline pageserver/src/tenant/storage_layer/layer.rs:1628]: This is the opposite of permanent, now all errors are transient.

File: pageserver/src/tenant/storage_layer/layer.rs [hunk 2]
```diff
@@ -1700,32 +1701,29 @@ impl DownloadedLayer {
...
+            .get_or_init(init)
+            .await
             .as_ref()
-            .map_err(|e| {
-                // errors are not clonabled, cannot but stringify
-                // test_broken_timeline matches this string
-                anyhow::anyhow!("layer loading failed: {e:#}")
-            })
+            // We already logged the full backtrace above, once. Don't repeat that here.
+            .map_err(|e| anyhow::anyhow!("layer load failed earlier: {e}"))
```
Comment (koivunej) [inline pageserver/src/tenant/storage_layer/layer.rs:1726]: why not

```suggestion
            .map_err(|e| anyhow::anyhow!("layer load failed earlier: {e:#}"))
```

`{e}` will be just the topmost.

**Review by koivunej — APPROVED** [review]

File: pageserver/src/tenant/storage_layer/layer.rs (hunk 2 above)
Comment (problame) [reply pageserver/src/tenant/storage_layer/layer.rs:1726]: It's only one level right now and I really want this error to be a summary of what happened earlier.

Comment (koivunej) [reply pageserver/src/tenant/storage_layer/layer.rs:1726]: oki

**Pull request merged**